README.md

```diff
@@ -101,12 +116,31 @@ git commit --amend                      # Re-do the commit message of the previo
 - or for instance Notepad++:
 `git config --global core.editor "'C:/Program Files/Notepad++/notepad++.exe' -multiInst -notabbar -nosession -noPlugin"`
 
+# Re/moving files under version control
+git rm <path/to/the/file>                 # remove file and stage the change to be committed.
+git mv <source/file> <destination/file>   # move/rename file and stage the change to be committed.  
 
 
 # See history
 git log             # Show commit logs
 git log --oneline   # Formats commits to a single line (shorthand for --pretty=oneline  --abbrev-commit )
 git log --graph     # Show a graph commits and branches
+git log --pretty=fuller     # To see commit log details with author and committer details, if any different.
+git log --follow <file>     # List history of given file.
+git log branch2..branch1    # Show commits present on branch1 compared to branch1
+
+# Deferring
+git stash                               # Stash (store temporarily) changes in working branch and enable checkingout a new branch
+git stash list                          # List stored stashes.
+git stash apply <stash>                 # Apply given <stash>, or if none given the latest from stash list.
+
+# Tagging
+
+
+git tag                                 # List tags
+git tag -l "name*"                      # List tags that are named <name>
+git tag -a "v1.0" -m "comment - v1.0"   # Creating annotated tag that creates checksum for the commit including verbose tagger details.
+git tag -d "v1.0"                       # Delete given tag in local repo
```

Comment:
Tagging is not really useful/needed at kata level (except in the tagging related katas where these commands are or should be mentioned).
Maybe it can be included in a separate CHEATSHEET.md but not here.

Reply:
Taken away.